plotly/tools.py

```diff
@@ -4559,6 +4563,23 @@ def _scatterplot_theme(dataframe, headers, diag, size, height,
                     return fig
 
     @staticmethod
+    def _hide_tick_labels_from_box_subplots(fig):
+        """
+        Hides tick labels for box plots in scatterplotmatrix subplots.
+        """
+        boxplot_xaxes = []
+        for j, plot_data in enumerate(fig['data']):
+            if plot_data['type'] == 'box':
+                boxplot_xaxes.append(
+                    'xaxis{}'.format(plot_data['xaxis'][-1])
+                )
+
+        for xaxis in boxplot_xaxes:
+            fig['layout'][xaxis]['showticklabels'] = False
+
+        return fig
```

Comment:
Personal preference, but I'm not a huge fan of the chaining pattern. You're mutating `fig` in this function and it makes it *super* clear that that's your intention if you don't return anything.

I.e., don't return `fig` to make it clear to callers that this function mutates the `fig` you give it.